src/service.rs

```diff
@@ -331,14 +338,19 @@ impl Account {
         let main_key = &self.main_key;
         let login_challenge = authenticator.get_login_challenge(username)?;
 
-        let old_main_crypto_manager = MainCryptoManager::new(try_into!(&main_key[..])?, version)?;
+        let old_main_crypto_manager = MainCryptoManager::new(main_key, version)?;
         let content = old_main_crypto_manager
             .0
             .decrypt(&self.user.encrypted_content, None)?;
         let old_login_crypto_manager = old_main_crypto_manager.login_crypto_manager()?;
 
-        let main_key = derive_key(&login_challenge.salt, new_password)?;
-        let main_crypto_manager = MainCryptoManager::new(try_into!(&main_key[..])?, version)?;
+        // Reuse the old salt if it's the right size, otherwise generate a new one
```

Comment:
Should we even be reusing the old salt?

Reply:
I don't think why it was done this way, maybe git blame can show the thinking (maybe on JS and not here).

I think there may have been a reason, though I'm unsure.